File: IsPrime/java/IsPrime.java
// Miller–Rabin primality test
import java.lang.*;

public class Main {
  public static boolean isPrime(int num){
    if ((num == 0) || (num == 1)){
      return true;
    }

    if ((num % 2 == 0) || (num % 3 == 0)){
      return false;
    }

    int sqrt_num = (int) Math.floor(Math.sqrt(num));

    // for i in range(sqrt_num, 1, -1):
    //     if num % i == 0:
    //         return False

    for (int i = sqrt_num; i > 2; i--){
      if (num % i == 0){
        return false;
      }
    }

    return false;
  }

  public static void main(String[] args) {
    System.out.println("isPrime(0) = " + isPrime(0));
    System.out.println("isPrime(1) = " + isPrime(1));
    System.out.println("isPrime(2) = " + isPrime(2));
    System.out.println("isPrime(3) = " + isPrime(3));
    System.out.println("isPrime(9) = " + isPrime(9));
    System.out.println("isPrime(49) = " + isPrime(49));
    System.out.println("isPrime(51) = " + isPrime(51));
    System.out.println("isPrime(7919) = " + isPrime(7919));
  }
}

File: IsPrime/python/IsPrime.py
# Miller–Rabin primality test
import math

def is_prime(num):
    if num == 0 or num == 1:
        return True
    
    if num % 2 == 0 or num % 3 == 0:
        return False

    sqrt_num = math.floor(math.sqrt(num))

    for i in range(sqrt_num, 1, -1):
        if num % i == 0:
            return False

    return True

def main():
    print('is_prime(0) = ' + str(is_prime(0)))
    print('is_prime(1) = ' + str(is_prime(1)))
    print('is_prime(2) = ' + str(is_prime(2)))
    print('is_prime(3) = ' + str(is_prime(3)))
    print('is_prime(9) = ' + str(is_prime(9)))
    print('is_prime(49) = ' + str(is_prime(49)))
    print('is_prime(51) = ' + str(is_prime(51)))
    print('is_prime(7919) = ' + str(is_prime(7919)))

if __name__ == "__main__":
    main()
